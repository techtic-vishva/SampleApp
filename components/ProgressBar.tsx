import { chattanoogaTapWater, dichotomousHippopotamus, green } from '../constants/Colors'
import { View } from './Themed'

export default function ({ count, max }: { count: number; max: number }) {
	const isComplete = count >= max
	return (
		<View
			style={{
				marginVertical: 3,
				backgroundColor: chattanoogaTapWater,
				height: 5,
				width: '100%',
				borderRadius: 20,
				flexDirection: 'row'
			}}>
			<View
				style={{
					width: `${(count / max) * 100}%`,
					backgroundColor: isComplete ? green : dichotomousHippopotamus,
					borderRadius: 20
				}}></View>
		</View>
	)
}
