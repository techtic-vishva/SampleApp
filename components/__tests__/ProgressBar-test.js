import * as React from 'react'
import renderer from 'react-test-renderer'
import ProgressBar from '../ProgressBar'

it('render when passing the props', () => {
	const tree = renderer.create(<ProgressBar count={0} max={100} />).toJSON()
	expect(tree).toMatchSnapshot()
})
