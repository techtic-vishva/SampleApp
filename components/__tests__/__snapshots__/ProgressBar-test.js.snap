// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render when passing the props 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#0b0805",
      },
      Object {
        "backgroundColor": "#999a81",
        "borderRadius": 20,
        "flexDirection": "row",
        "height": 5,
        "marginVertical": 3,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#0b0805",
        },
        Object {
          "backgroundColor": "#ed832d",
          "borderRadius": 20,
          "width": "0%",
        },
      ]
    }
  />
</View>
`;
